course/migrations/0047_delete_duplicate_enrollments.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 2.2.13 on 2020-06-26 07:55
+
+from django.db import migrations
+from django.db.models import Count, Min
+from course.models import Enrollment
+
+
+def delete_duplicates(apps, schema_editor):
+
+    unique_fields = ['course_instance', 'user_profile']
+
+    duplicates = (
+        Enrollment.objects.values(*unique_fields)
+        .order_by()
+        .annotate(min_id=Min('id'), count_id=Count('id'))
+        .filter(count_id__gt=1)
+        )
```

Comment:
I wanted to say here, that I'm considering which style should be suggested for code like this. Based ond PEP 8, current layout is fine. Though, I typically prefer closing bracket to be in the same column as the starting line (e.g. here at the start of the line).

For discussion, here are relevant layout posibilities:

```python
# pros: clean and compact, and subject of the action is in the first line
# cons: problematic when adding a new method
duplicates = (Enrollment.objects
    .values(*unique_fields)
    .order_by()
    .annotate(min_id=Min('id'), count_id=Count('id'))
    .filter(count_id__gt=1))

# pros: closing bracket is in a right place, and subject of the action is in the first line
# cons: starting and closing brackets don't feel to be in balance
duplicates = (Enrollment.objects
    .values(*unique_fields)
    .order_by()
    .annotate(min_id=Min('id'), count_id=Count('id'))
    .filter(count_id__gt=1)
)

# pros: newline after the starting bracket
# cons: no separation between object and chained methods
duplicates = (
    Enrollment.objects
    .values(*unique_fields)
    .order_by()
    .annotate(min_id=Min('id'), count_id=Count('id'))
    .filter(count_id__gt=1)
)

# pros: newline in start, and methods are indented to separate from the object
# cons: takes a lot of space
duplicates = (
    Enrollment.objects
        .values(*unique_fields)
        .order_by()
        .annotate(min_id=Min('id'), count_id=Count('id'))
        .filter(count_id__gt=1)
)
```

As comparison, I would consider the following example to match Airbnb's JavaScript style guide.

```javascript
const selected = (
  $('#items')
    .find('.open')
      .highlight()
      .end()
    .find(':selected')
      .first()
)
```

This would give merit for the last python layout. Anyone, feel free to write opinions here.

Reply:
I like the enrollment options 3 and 4.

Reply:
I like options 3 and 4 the most as well. 4 is my favourite and I think it's the easiest to read.

Reply:
I think the last one would be the one to recommend.

Reply:
Let's go with the last one then.